[BUG] Issue with typechecking component testing with "moduleResolution": bundler

<!-- ⚠️⚠️ Do not delete this template ⚠️⚠️ -->

<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Playwright release to see if your issue has already been fixed -->
<!-- 💡 Provide enough information for us to be able to reproduce your issue locally -->

### System info
- Playwright Version: v1.35
- Operating System: not applicable (MacOS 13.3)
- Browser: not applicable

### Problem

I'm converting the Typescript configuration of my project from `moduleResolution: "node"` to `moduleResolution: "bundler"`, it mostly works but typechecking the project reports one error in `node_modules/@playwright`:

```
node_modules/@playwright/experimental-ct-react/index.d.ts:26:33 - error TS2307: Cannot find module '@playwright/experimental-ct-core/types/component' or its corresponding type declarations.

26 import type { JsonObject } from '@playwright/experimental-ct-core/types/component';
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I would guess that the problem is that `types/component` is not mentioned in the `exports` field of `experimental-ct-core/package.json`, so importing from it does not work. I verified that it works if I replace the offending line by the line below, but I don’t know if it is the right thing to do:

```ts
import type { JsonObject } from '../experimental-ct-core/types/component';
```

It works fine with `moduleResolution: "node"`, but from what I understand that's because Typescript then ignores the `exports` field of `package.json`.

### Source code
- [X] I provided exact source code that allows reproducing the issue locally.

To reproduce, create a directory with the following three files, then run `npm install` and `npx tsc`.

`package.json`:
```json
{
  "name": "playwright-ts-issue",
  "devDependencies": {
    "@playwright/experimental-ct-react": "^1.35.1",
    "@types/react": "^18.2.14",
    "typescript": "^5.1.6"
  }
}
```

`tsconfig.json`:
```json
{
  "compilerOptions": {
    "noEmit": true,
    "module": "esnext",
    "moduleResolution": "bundler",
  },
  "include": ["./index.ts"],
}
```

`index.ts`:
```js
import "@playwright/experimental-ct-react";
```

--- a/packages/playwright-ct-core/index.d.ts
+++ b/packages/playwright-ct-core/index.d.ts
@@ -23,7 +23,7 @@ import type {
   PlaywrightWorkerOptions,
   Locator,
 } from '@playwright/test';
-import type { JsonObject } from '@playwright/experimental-ct-core/types/component';
+import type { JsonObject } from './types/component';
 import type { InlineConfig } from 'vite';
 
 export type PlaywrightTestConfig<T = {}, W = {}> = Omit<BasePlaywrightTestConfig<T, W>, 'use'> & {

--- a/packages/playwright-ct-core/package.json
+++ b/packages/playwright-ct-core/package.json
@@ -18,7 +18,10 @@
     },
     "./cli": "./cli.js",
     "./lib/mount": "./lib/mount.js",
-    "./lib/vitePlugin": "./lib/vitePlugin.js"
+    "./lib/vitePlugin": "./lib/vitePlugin.js",
+    "./types/component": {
+      "types": "./types/component.d.ts"
+    }
   },
   "dependencies": {
     "playwright-core": "1.38.0-next",

